### DISCUSSION TITLE
Seeking help with firmware issues with EFR32MGM210PA32JIA2

### DISCUSSION THREAD

**Discussion by signum2023** [Koenkk/zigbee2mqtt#18072] [Devices]
I am using ZigBee2MQTT with an MGM210PA32JIA2 Zigbee module connected via the UART interface (UART NCP at 57600baud with SoftwareFlowControl XonXoff).

I have tested ZigBee2MQTT (v1.25.1/1.30.2/1.31.2) with different firmware versions to assess range and stability, and here's what I have observed:

Firmware version 6.10.3.0 proved to be particularly stable but had a very limited range.

On the other hand, firmware versions 7.2.3 (EZSP v11) and 7.3.00 (EZSP v12) offered excellent range. However, I encountered crashes when attempting to control multiple devices simultaneously.

I am seeking a stable solution! Therefore, I would like to stick with firmware version 6.10.3 (EZSP v8), but it doesn't make sense without reasonable range. I'm wondering why this version has such poor range. Could it be due to EZSP v8?

Alternatively, why are firmware versions 7.2.3 (EZSP v11) and 7.3.0 (EZSP v12) unstable when attempting to simultaneously control a range of lamps?

I would greatly appreciate any hints or suggestions for a solution.

**Comment by signum2023** [comment]
Found the problem and compiled new firmware

**Reply by ra-kal** [reply]
I'm having stability issues with Sonoff Dongle-E. My network is not reliable, e.g. once in 10 times a button press is not registered or command not received by a end device. I live in a city with a lot of WiFI networks around me. I recently changed channel from 11 to 25 in a hope it will be better, but there was no change in reliability at all. My problem is I do not know what is causing the sudden dropouts, if it is the coordinator, or some devices, or the WiFi network, or communication with Z2M.

I have not experiented with various FW versions as nobody could tell me, what are the differences and why we should stick with the "old" 6.10.3.

You have written, you found the problem and compiled new FW. Does the new FW versions improve stability / reliability? What version would you recommend for the Sonoff Dongle-E?
